router/router.go

```diff
@@ -1805,22 +1808,21 @@ func (rt *HandleT) collectMetrics(ctx context.Context) {
 		// failureMetric struct
 		rt.failureMetricLock.RLock()
```

Comment:
A full lock would be appropriate here, since at the end we are ressetting the `failuresMetric` map